### ISSUE TITLE
New CloudWatch metric requested - `RecordsSuccessfullySent`

### ISSUE THREAD

**Issue by diranged** [awslabs/amazon-kinesis-agent#82]
Right now you record the number of `RecordSendAttempts` and `RecordSendErrors`. However, these two cannot be used to accurately gauge exactly how many events _have been sent_ to Kinesis.

Based on my read of the code,`RecordSendAttempts` increases with every single attempt to push records, regardless of whether or not the records succeeded. `RecordSendErrors` will record the number of those records that did not actually succeed, as long as the API returned properly. However, https://github.com/awslabs/amazon-kinesis-agent/blob/51c60e07b7f2069a1424aa651fa09a2b46fdc721/src/com/amazon/kinesis/streaming/agent/tailing/KinesisSender.java#L101-L108 implies that if there is an underlying service error (say we're being throttled?), that the function exits and ends up being re-run at a later point. When this happens, the `RecordSendAttempts` metric is invalid because it will record multiple send attempts.

One way to fix this, I think, is to move https://github.com/awslabs/amazon-kinesis-agent/blob/51c60e07b7f2069a1424aa651fa09a2b46fdc721/src/com/amazon/kinesis/streaming/agent/tailing/KinesisSender.java#L98 to line #99 or #100. However, then the metric name doesn't make sense.

Another way to fix this is to add another metric `RecordsSuccessfullySent` at https://github.com/awslabs/amazon-kinesis-agent/blob/51c60e07b7f2069a1424aa651fa09a2b46fdc721/src/com/amazon/kinesis/streaming/agent/tailing/KinesisSender.java#L120.

(Note, I am also still wondering if you're interested in doing #46)

**Comment by chaochenq** [comment]
Hi @diranged you are correct. We made the metrics like this mostly because we wanted the metrics to monitor the health of the tailing of the Agent. As you said, RecordSendAttempts count retry as well. And RecordSendErrors count throttled records.

I guess the absolute value of these metrics won't make as much sense as the baseline change. I.e. 

What if BytesSent has spiked and RecordSendAttempts is unchanged? (maybe because your record doesn't have a valid terminator or a different one than what you specified in agent config) 
What if RecordSendErrors has spiked and BytesSent is unchanged? (seems like throttling issue)
What if RecordSendErrors has spiked and BytesSent has dropped? (maybe non-throttling issue)
...

So I think the existing metrics including retries would still be useful. Adding a new metric of RecordsSuccessfullySent sounds good to me.

**Comment by diranged** [comment]
Yeah I'm not knocking the current metrics -- just hoping to add some more. Also, _host specific_ tags are critical here to actually being able to track down where something bad is happening.
